### PR TITLE
EE Cache: Fix cache masking, support caching invalid phadr, and cache line locking

### PR DESCRIPTION
**Pull request by F0bes**
### Description of Changes
You are able to make TLB entries that point to invalid physical addresses. Because the cache is accessed before the physical write , the cache will still store data for these physical addresses. Once the cache data is evicted, curiously there is no bus error for addresses such as 0x70004000, so the data is silently discarded.
Find my Own Way uses this method to create a "virtual scratchpad" of higher speed, temporary memory.

Previously our cache line locking wasn't fully implemented. This PR addresses that as well.
With the combined invalid physical address mapping and cache line locking support, **I believe I have fixed the find my own way demo!.**

### Rationale behind Changes
This is proper cache behaviour.

### Suggested Testing Steps
Test games that require EE Cache (I'm sorry, I know it's slow :( ), make sure they still work.
Also, **if any game causes bus errors** please try and test it with this PR and EE Cache enabled. This might be the fix required.

## If you're testing games with bus errors (including find my own way), I recommend disabling the system console. The bus error spam slows it down.

### CODE REVIEW

File: pcsx2/vtlb.cpp [hunk 1]
```diff
@@ -559,12 +567,35 @@ static void vtlbUnmappedVWriteSm(u32 addr, OperandType data) { vtlb_Miss(addr, 1
 static void TAKES_R128 vtlbUnmappedVWriteLg(u32 addr, r128 data) { vtlb_Miss(addr, 1); }
 
 template <typename OperandType>
-static OperandType vtlbUnmappedPReadSm(u32 addr) { vtlb_BusError(addr, 0); return 0; }
-static RETURNS_R128 vtlbUnmappedPReadLg(u32 addr) { vtlb_BusError(addr, 0); return r128_zero(); }
+static OperandType vtlbUnmappedPReadSm(u32 addr) {
+	vtlb_BusError(addr, 0);
+	if(!CHECK_EEREC && CHECK_CACHE && CheckCache(addr)){
+		switch (sizeof(OperandType)) {
```
Comment (lightningterror) [inline pcsx2/vtlb.cpp:573]: Can you clang format the changes so the formatting is consistent, can also do it on the entire file.

**Comment by F0bes** [comment]
Going to look into properly locking cache lines.

**Comment by F0bes** [comment]
@MrCK1 If you're going to accidentally mark the PR as ready, you could've put it back as a draft lol.

I've updated the PR, it's ready on my end. I'm confident I fixed find my own way. It's hard to know though, it's been running for 20 minutes at 1-3 FPS 🫠.

**Comment by refractionpcsx2** [comment]
LGTM, maybe needs testing on linux to make sure the popcount thing doesn't explode :D

Tested on windows with find my own way + dead or alive with patches disabled

**Comment by kamfretoz** [comment]
Did somebody say Linux? 👀

**Review by Mrlinkwii — APPROVED** [review]

**Review by kamfretoz — APPROVED** [review]

**Pull request merged**